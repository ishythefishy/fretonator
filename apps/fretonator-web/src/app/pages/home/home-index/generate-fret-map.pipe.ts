import { Pipe, PipeTransform } from '@angular/core';
import { FretMap, Mode, NoteObject } from '../../../util/types';
import { FretMapService } from '../../../common/fret-map/fret-map.service';

@Pipe({
  name: 'generateFretMap',
})
export class GenerateFretMapPipe implements PipeTransform {
  constructor(public generateFretMapService: FretMapService) {}

  transform(note: NoteObject, mode: Mode): FretMap {
    return this.generateFretMapService.getFretMapping(note, mode);
  }
}
