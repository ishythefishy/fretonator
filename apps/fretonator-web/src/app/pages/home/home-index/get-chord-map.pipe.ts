import { Pipe, PipeTransform } from '@angular/core';
import { FretMapService } from '../../../common/fret-map/fret-map.service';
import { ChordMap, Mode, NoteObject } from '../../../util/types';

@Pipe({
  name: 'getChordMap'
})
export class GetChordMapPipe implements PipeTransform {

  constructor(public generateFretMapService: FretMapService) {
  }

  transform(noteObject: NoteObject, mode: Mode): ChordMap {
    return this.generateFretMapService.getChordMap(noteObject, mode);
  }

}
