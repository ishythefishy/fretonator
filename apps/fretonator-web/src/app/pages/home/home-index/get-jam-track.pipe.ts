import { Pipe, PipeTransform } from '@angular/core';
import { FretMapService } from '../../../common/fret-map/fret-map.service';
import { JamTrack, Mode, NoteObject } from '../../../util/types';

@Pipe({
  name: 'getJamTrack',
})
export class GetJamTrackPipe implements PipeTransform {
  constructor(public generateFretMapService: FretMapService) {}

  transform(note: NoteObject, mode: Mode): JamTrack | false {
    return this.generateFretMapService.getJamTrack(note, mode);
  }
}
