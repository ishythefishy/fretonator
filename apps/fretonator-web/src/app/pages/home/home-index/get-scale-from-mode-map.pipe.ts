import { Pipe, PipeTransform } from '@angular/core';
import { Mode, NoteObject, Scale } from '../../../util/types';
import { FretMapService } from '../../../common/fret-map/fret-map.service';

@Pipe({
  name: 'getScaleFromModeMap',
})
export class GetScaleFromModeMapPipe implements PipeTransform {
  constructor(private generateFretMapService: FretMapService) {
  }

  transform(note: NoteObject, mode: Mode): Scale {
    const modeMap = this.generateFretMapService.generateMode(note, mode);
    return modeMap.map(thisNote => this.generateFretMapService.convertNoteObjectToHumanReadable(thisNote));
  }
}
