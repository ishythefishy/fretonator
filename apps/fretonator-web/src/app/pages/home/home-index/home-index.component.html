<main>
  <div class="hero">
    <div class="hero__inner">
      <h1 class="hero__title">
        <span *ngIf="showHowTo">How to use the Fretonator</span>
        <span *ngIf="!showHowTo">Fretonator: the ultimate interactive free guitar theory tool</span>
      </h1>
      <ol class="hero__list"
          id="howToList"
          [class.hidden]="!showHowTo"
          [attr.aria-expanded]="showHowTo">
        <li class="hero__text hero__text--peach" data-step="1">Choose a starting note</li>
        <li class="hero__text hero__text--yellow" data-step="2">Pick a mode</li>
        <li class="hero__text hero__text--purple" data-step="3">Check out the fretboard</li>
        <li class="hero__text hero__text--turq" data-step="4">Have a jam!</li>
      </ol>
      <button class="hero__toggleHowTo"
              [attr.aria-checked]="showHowTo"
              aria-label="Toggle visibility of Fretonator guide"
              aria-controls="howToList"
              role="switch"
              type="button"
              (click)="toggleHowTo()">
        <span *ngIf="showHowTo">Hide</span>
        <span *ngIf="!showHowTo">Show how-to</span>
      </button>
    </div>
  </div>

  <div class="modeSelector__container">
    <div class="modeSelector__chipsContainer">
      <h2 class="modeSelector__label">Starting note</h2>

      <app-chips>
        <app-chip *ngFor="let thisNote of octave"
                  [note]="thisNote"
                  [noteExtender]="noteExtenderString"
                  [mode]="mode"
                  rounded="true"
                  selectedColor="default">{{thisNote | titlecase}}</app-chip>
      </app-chips>

      <app-chips>
        <app-chip *ngIf="!(note === 'f' || note === 'c')"
                  [note]="note"
                  [noteExtender]="'flat'"
                  [mode]="mode"
                  rounded="true"
                  selectedColor="grey">{{note | titlecase}}â™­
        </app-chip>

        <app-chip [note]="note"
                  [noteExtender]="'natural'"
                  [mode]="mode"
                  rounded="true"
                  selectedColor="grey">{{note | titlecase}}</app-chip>

        <app-chip *ngIf="!(note === 'b' || note === 'e')"
                  [note]="note"
                  [noteExtender]="'sharp'"
                  [mode]="mode"
                  rounded="true"
                  selectedColor="grey">{{note | titlecase}}#
        </app-chip>
      </app-chips>
    </div>

    <div class="modeSelector__chipsContainer">
      <h2 class="modeSelector__label">Mode</h2>

      <app-chips>
        <app-chip *ngFor="let thisMode of modeSelectorObjects"
                  [note]="note"
                  [noteExtender]="noteExtenderString"
                  [mode]="thisMode.name"
                  selectedColor="default">{{thisMode.displayName}}</app-chip>
      </app-chips>

    </div>

    <app-fretonator [scale]="note | symbolToNoteObj: noteExtender | getScaleFromModeMap: mode"
                    [fretMap]="note | symbolToNoteObj: noteExtender | generateFretMap: mode"
                    [intervalPattern]="mode | getIntervalPatternFromMode"
                    [chordMap]="note | symbolToNoteObj: noteExtender | getChordMap: mode"
                    [mode]="mode"></app-fretonator>

    <ng-container *ngIf="note | symbolToNoteObj: noteExtender | getJamTrack: mode">
      <app-video-loader [jamTrack]="note | symbolToNoteObj: noteExtender | getJamTrack: mode"></app-video-loader>
    </ng-container>

  </div>
</main>
