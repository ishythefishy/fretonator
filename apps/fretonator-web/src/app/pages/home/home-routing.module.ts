import { NgModule } from '@angular/core';
import { Route, RouterModule, Routes } from '@angular/router';
import { HomeIndexComponent } from './home-index/home-index.component';
import { RouteToModeConfigurationService } from '../../common/route-to-mode-configuration/route-to-mode-configuration.service';
import { IsValidModeConfigurationGuard } from '../../common/is-valid-mode-configuration/is-valid-mode-configuration.guard';

const home: Route = {
  path: '',
  pathMatch: 'full',
  resolve: {
    selected: RouteToModeConfigurationService,
  },
  component: HomeIndexComponent,
};

const fretonator: Route = {
  path: ':note/:noteExtender/:mode',
  resolve: {
    selected: RouteToModeConfigurationService,
  },
  canActivate: [
    IsValidModeConfigurationGuard
  ],
  component: HomeIndexComponent,
};

const routes: Routes = [home, fretonator];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule],
})
export class HomeRoutingModule {}
