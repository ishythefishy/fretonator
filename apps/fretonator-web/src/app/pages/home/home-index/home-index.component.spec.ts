import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { HomeIndexComponent } from './home-index.component';
import { Component } from '@angular/core';
import { HomeModule } from '../home.module';
import { By } from '@angular/platform-browser';
import { BrowserTestingModule } from '@angular/platform-browser/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { ActivatedRoute } from '@angular/router';
import { of } from 'rxjs';

describe('HomeIndexComponent', () => {
  const selectors = {
    howToToggle: By.css('.hero__toggleHowTo'),
    howToList: By.css('.hero__list')
  };

  const classNames = {
    hideHowTo: 'hidden'
  };

  @Component({
    selector: 'app-home-index-spec',
    template: `
      <app-home-index></app-home-index>
    `
  })
  class HomeIndexComponentSpec {
  }

  let component: HomeIndexComponentSpec;
  let fixture: ComponentFixture<HomeIndexComponentSpec>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [HomeIndexComponentSpec],
      imports: [
        HomeModule,
        BrowserTestingModule,
        NoopAnimationsModule,
        RouterTestingModule
      ],
      providers: [
        {
          provide: ActivatedRoute,
          useValue: {
            params: of({}),
            snapshot: {
              data: {
                selected: { note: 'c', noteExtender: 'natural', mode: 'ionian' }
              }
            }
          }
        }
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(HomeIndexComponentSpec);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should toggle the how-to section', () => {
    const howToToggle = fixture.debugElement.query(selectors.howToToggle);
    const howToList = fixture.debugElement.query(selectors.howToList);
    howToToggle.nativeElement.click();
    fixture.detectChanges();
    expect(howToList.classes[classNames.hideHowTo]).toBeTruthy();
  });
});
