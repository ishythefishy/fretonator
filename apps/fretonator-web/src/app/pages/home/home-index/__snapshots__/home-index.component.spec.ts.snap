// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeIndexComponent should create 1`] = `
<app-home-index-spec>
  <app-home-index>
    <main>
      <div
        class="hero"
      >
        <div
          class="hero__inner"
        >
          <h1
            class="hero__title"
          >

            <span>
              How to use the Fretonator
            </span>

          </h1>
          <ol
            aria-expanded="true"
            class="hero__list"
            id="howToList"
          >
            <li
              class="hero__text hero__text--peach"
              data-step="1"
            >
              Choose a starting note
            </li>
            <li
              class="hero__text hero__text--yellow"
              data-step="2"
            >
              Pick a mode
            </li>
            <li
              class="hero__text hero__text--purple"
              data-step="3"
            >
              Check out the fretboard
            </li>
            <li
              class="hero__text hero__text--turq"
              data-step="4"
            >
              Have a jam!
            </li>
          </ol>
          <button
            aria-checked="true"
            aria-controls="howToList"
            aria-label="Toggle visibility of Fretonator guide"
            class="hero__toggleHowTo"
            role="switch"
            type="button"
          >

            <span>
              Hide
            </span>

          </button>
        </div>
      </div>
      <div
        class="modeSelector__container"
      >
        <div
          class="modeSelector__chipsContainer"
        >
          <h2
            class="modeSelector__label"
          >
            Starting note
          </h2>
          <app-chips>
            <div
              class="chips"
            >

              <app-chip
                rounded="true"
                selectedcolor="default"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip chip--round"
                  href="/a/natural/ionian"
                >
                  A
                </a>
              </app-chip>
              <app-chip
                rounded="true"
                selectedcolor="default"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip chip--round"
                  href="/b/natural/ionian"
                >
                  B
                </a>
              </app-chip>
              <app-chip
                rounded="true"
                selectedcolor="default"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip chip--active chip--round"
                  href="/c/natural/ionian"
                >
                  C
                </a>
              </app-chip>
              <app-chip
                rounded="true"
                selectedcolor="default"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip chip--round"
                  href="/d/natural/ionian"
                >
                  D
                </a>
              </app-chip>
              <app-chip
                rounded="true"
                selectedcolor="default"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip chip--round"
                  href="/e/natural/ionian"
                >
                  E
                </a>
              </app-chip>
              <app-chip
                rounded="true"
                selectedcolor="default"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip chip--round"
                  href="/f/natural/ionian"
                >
                  F
                </a>
              </app-chip>
              <app-chip
                rounded="true"
                selectedcolor="default"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip chip--round"
                  href="/g/natural/ionian"
                >
                  G
                </a>
              </app-chip>
            </div>
          </app-chips>
          <app-chips>
            <div
              class="chips"
            >

              <app-chip
                rounded="true"
                selectedcolor="grey"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip chip--active chip--round chip--grey"
                  href="/c/natural/ionian"
                >
                  C
                </a>
              </app-chip>

              <app-chip
                rounded="true"
                selectedcolor="grey"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip chip--round chip--grey"
                  href="/c/sharp/ionian"
                >
                  C#
                </a>
              </app-chip>
            </div>
          </app-chips>
        </div>
        <div
          class="modeSelector__chipsContainer"
        >
          <h2
            class="modeSelector__label"
          >
            Mode
          </h2>
          <app-chips>
            <div
              class="chips"
            >

              <app-chip
                selectedcolor="default"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip chip--active"
                  href="/c/natural/ionian"
                >
                  Ionian (Major)
                </a>
              </app-chip>
              <app-chip
                selectedcolor="default"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip"
                  href="/c/natural/dorian"
                >
                  Dorian
                </a>
              </app-chip>
              <app-chip
                selectedcolor="default"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip"
                  href="/c/natural/phrygian"
                >
                  Phrygian
                </a>
              </app-chip>
              <app-chip
                selectedcolor="default"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip"
                  href="/c/natural/lydian"
                >
                  Lydian
                </a>
              </app-chip>
              <app-chip
                selectedcolor="default"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip"
                  href="/c/natural/mixolydian"
                >
                  Mixolydian
                </a>
              </app-chip>
              <app-chip
                selectedcolor="default"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip"
                  href="/c/natural/aolian"
                >
                  Aolian (Minor)
                </a>
              </app-chip>
              <app-chip
                selectedcolor="default"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip"
                  href="/c/natural/locrian"
                >
                  Locrian
                </a>
              </app-chip>
              <app-chip
                selectedcolor="default"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip"
                  href="/c/natural/harmonicMinor"
                >
                  Harmonic Minor
                </a>
              </app-chip>
              <app-chip
                selectedcolor="default"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip"
                  href="/c/natural/phrygianDominant"
                >
                  Phrygian Dominant
                </a>
              </app-chip>
              <app-chip
                selectedcolor="default"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip"
                  href="/c/natural/majorPentatonic"
                >
                  Major Pentatonic
                </a>
              </app-chip>
              <app-chip
                selectedcolor="default"
              >
                <a
                  aria-label="note | createChipLink: noteExtender: mode"
                  class="chip"
                  href="/c/natural/minorPentatonic"
                >
                  Minor Pentatonic
                </a>
              </app-chip>
            </div>
          </app-chips>
        </div>
        <app-fretonator>

          <div
            class="arrowHint"
          >
            <span
              class="arrowHint__arrow"
            />
          </div>
          <div
            class="fretboard__container"
          >
            <div
              class="fretboard"
              data-fretboard=""
            >



              <div
                class="fretboard__cell fretboard__cell--string fretboard__cell--selected"
                data-degree="mediant"
                data-display-note="E"
                data-fret="0"
                data-mode="ionian"
                data-string="E"
                data-string-name="e"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="subdominant"
                data-display-note="F"
                data-fret="1"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell"
                data-fret="2"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="dominant"
                data-display-note="G"
                data-fret="3"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell"
                data-fret="4"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="submediant"
                data-display-note="A"
                data-fret="5"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell"
                data-fret="6"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="leadingTone"
                data-display-note="B"
                data-fret="7"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="tonic"
                data-display-note="C"
                data-fret="8"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell"
                data-fret="9"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="supertonic"
                data-display-note="D"
                data-fret="10"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell"
                data-fret="11"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="mediant"
                data-display-note="E"
                data-fret="12"
                data-mode="ionian"
                data-string="E"
              />




              <div
                class="fretboard__cell fretboard__cell--string fretboard__cell--selected"
                data-degree="leadingTone"
                data-display-note="B"
                data-fret="0"
                data-mode="ionian"
                data-string="B"
                data-string-name="B"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="tonic"
                data-display-note="C"
                data-fret="1"
                data-mode="ionian"
                data-string="B"
              />

              <div
                class="fretboard__cell"
                data-fret="2"
                data-mode="ionian"
                data-string="B"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="supertonic"
                data-display-note="D"
                data-fret="3"
                data-mode="ionian"
                data-string="B"
              />

              <div
                class="fretboard__cell"
                data-fret="4"
                data-mode="ionian"
                data-string="B"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="mediant"
                data-display-note="E"
                data-fret="5"
                data-mode="ionian"
                data-string="B"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="subdominant"
                data-display-note="F"
                data-fret="6"
                data-mode="ionian"
                data-string="B"
              />

              <div
                class="fretboard__cell"
                data-fret="7"
                data-mode="ionian"
                data-string="B"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="dominant"
                data-display-note="G"
                data-fret="8"
                data-mode="ionian"
                data-string="B"
              />

              <div
                class="fretboard__cell"
                data-fret="9"
                data-mode="ionian"
                data-string="B"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="submediant"
                data-display-note="A"
                data-fret="10"
                data-mode="ionian"
                data-string="B"
              />

              <div
                class="fretboard__cell"
                data-fret="11"
                data-mode="ionian"
                data-string="B"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="leadingTone"
                data-display-note="B"
                data-fret="12"
                data-mode="ionian"
                data-string="B"
              />




              <div
                class="fretboard__cell fretboard__cell--string fretboard__cell--selected"
                data-degree="dominant"
                data-display-note="G"
                data-fret="0"
                data-mode="ionian"
                data-string="G"
                data-string-name="G"
              />

              <div
                class="fretboard__cell"
                data-fret="1"
                data-mode="ionian"
                data-string="G"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="submediant"
                data-display-note="A"
                data-fret="2"
                data-mode="ionian"
                data-string="G"
              />

              <div
                class="fretboard__cell"
                data-fret="3"
                data-mode="ionian"
                data-string="G"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="leadingTone"
                data-display-note="B"
                data-fret="4"
                data-mode="ionian"
                data-string="G"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="tonic"
                data-display-note="C"
                data-fret="5"
                data-mode="ionian"
                data-string="G"
              />

              <div
                class="fretboard__cell"
                data-fret="6"
                data-mode="ionian"
                data-string="G"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="supertonic"
                data-display-note="D"
                data-fret="7"
                data-mode="ionian"
                data-string="G"
              />

              <div
                class="fretboard__cell"
                data-fret="8"
                data-mode="ionian"
                data-string="G"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="mediant"
                data-display-note="E"
                data-fret="9"
                data-mode="ionian"
                data-string="G"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="subdominant"
                data-display-note="F"
                data-fret="10"
                data-mode="ionian"
                data-string="G"
              />

              <div
                class="fretboard__cell"
                data-fret="11"
                data-mode="ionian"
                data-string="G"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="dominant"
                data-display-note="G"
                data-fret="12"
                data-mode="ionian"
                data-string="G"
              />




              <div
                class="fretboard__cell fretboard__cell--string fretboard__cell--selected"
                data-degree="supertonic"
                data-display-note="D"
                data-fret="0"
                data-mode="ionian"
                data-string="D"
                data-string-name="D"
              />

              <div
                class="fretboard__cell"
                data-fret="1"
                data-mode="ionian"
                data-string="D"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="mediant"
                data-display-note="E"
                data-fret="2"
                data-mode="ionian"
                data-string="D"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="subdominant"
                data-display-note="F"
                data-fret="3"
                data-mode="ionian"
                data-string="D"
              />

              <div
                class="fretboard__cell"
                data-fret="4"
                data-mode="ionian"
                data-string="D"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="dominant"
                data-display-note="G"
                data-fret="5"
                data-mode="ionian"
                data-string="D"
              />

              <div
                class="fretboard__cell"
                data-fret="6"
                data-mode="ionian"
                data-string="D"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="submediant"
                data-display-note="A"
                data-fret="7"
                data-mode="ionian"
                data-string="D"
              />

              <div
                class="fretboard__cell"
                data-fret="8"
                data-mode="ionian"
                data-string="D"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="leadingTone"
                data-display-note="B"
                data-fret="9"
                data-mode="ionian"
                data-string="D"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="tonic"
                data-display-note="C"
                data-fret="10"
                data-mode="ionian"
                data-string="D"
              />

              <div
                class="fretboard__cell"
                data-fret="11"
                data-mode="ionian"
                data-string="D"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="supertonic"
                data-display-note="D"
                data-fret="12"
                data-mode="ionian"
                data-string="D"
              />




              <div
                class="fretboard__cell fretboard__cell--string fretboard__cell--selected"
                data-degree="submediant"
                data-display-note="A"
                data-fret="0"
                data-mode="ionian"
                data-string="A"
                data-string-name="A"
              />

              <div
                class="fretboard__cell"
                data-fret="1"
                data-mode="ionian"
                data-string="A"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="leadingTone"
                data-display-note="B"
                data-fret="2"
                data-mode="ionian"
                data-string="A"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="tonic"
                data-display-note="C"
                data-fret="3"
                data-mode="ionian"
                data-string="A"
              />

              <div
                class="fretboard__cell"
                data-fret="4"
                data-mode="ionian"
                data-string="A"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="supertonic"
                data-display-note="D"
                data-fret="5"
                data-mode="ionian"
                data-string="A"
              />

              <div
                class="fretboard__cell"
                data-fret="6"
                data-mode="ionian"
                data-string="A"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="mediant"
                data-display-note="E"
                data-fret="7"
                data-mode="ionian"
                data-string="A"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="subdominant"
                data-display-note="F"
                data-fret="8"
                data-mode="ionian"
                data-string="A"
              />

              <div
                class="fretboard__cell"
                data-fret="9"
                data-mode="ionian"
                data-string="A"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="dominant"
                data-display-note="G"
                data-fret="10"
                data-mode="ionian"
                data-string="A"
              />

              <div
                class="fretboard__cell"
                data-fret="11"
                data-mode="ionian"
                data-string="A"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="submediant"
                data-display-note="A"
                data-fret="12"
                data-mode="ionian"
                data-string="A"
              />




              <div
                class="fretboard__cell fretboard__cell--string fretboard__cell--selected"
                data-degree="mediant"
                data-display-note="E"
                data-fret="0"
                data-mode="ionian"
                data-string="E"
                data-string-name="E"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="subdominant"
                data-display-note="F"
                data-fret="1"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell"
                data-fret="2"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="dominant"
                data-display-note="G"
                data-fret="3"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell"
                data-fret="4"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="submediant"
                data-display-note="A"
                data-fret="5"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell"
                data-fret="6"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="leadingTone"
                data-display-note="B"
                data-fret="7"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="tonic"
                data-display-note="C"
                data-fret="8"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell"
                data-fret="9"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="supertonic"
                data-display-note="D"
                data-fret="10"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell"
                data-fret="11"
                data-mode="ionian"
                data-string="E"
              />

              <div
                class="fretboard__cell fretboard__cell--selected"
                data-degree="mediant"
                data-display-note="E"
                data-fret="12"
                data-mode="ionian"
                data-string="E"
              />

            </div>
          </div>
          <div
            class="infoContainer"
          >
            <div
              class="infoContainer__row"
            >

              <div
                class="degrees"
              >
                <h4
                  class="degreesDisplay__title"
                >
                  Degrees
                </h4>
                <div
                  class="degreesDisplay"
                >
                  <span
                    class="degreesDisplay__degree degreesDisplay__degree--tonic"
                  >
                    1
                  </span>
                  <span
                    class="degreesDisplay__degree degreesDisplay__degree--mediant"
                  >
                    3
                  </span>
                  <span
                    class="degreesDisplay__degree degreesDisplay__degree--dominant"
                  >
                    5
                  </span>
                </div>
                <button
                  aria-checked="false"
                  aria-controls="scaleDegreesInfo"
                  aria-label="Scale Degrees Info Toggle"
                  class="scaleDegrees__toggle"
                  role="switch"
                  type="button"
                >
                  What is this?
                </button>
              </div>
              <div
                class="scaleDisplay"
              >
                <h4
                  class="scaleDisplay__title"
                >
                  Scale
                </h4>
                <div
                  class="scaleDisplay__notes"
                >


                  <h5
                    class="scaleDisplay__note"
                  >
                     C

                    <span>
                       -
                    </span>
                  </h5>

                  <h5
                    class="scaleDisplay__note"
                  >
                     D

                    <span>
                       -
                    </span>
                  </h5>

                  <h5
                    class="scaleDisplay__note"
                  >
                     E

                    <span>
                       -
                    </span>
                  </h5>

                  <h5
                    class="scaleDisplay__note"
                  >
                     F

                    <span>
                       -
                    </span>
                  </h5>

                  <h5
                    class="scaleDisplay__note"
                  >
                     G

                    <span>
                       -
                    </span>
                  </h5>

                  <h5
                    class="scaleDisplay__note"
                  >
                     A

                    <span>
                       -
                    </span>
                  </h5>

                  <h5
                    class="scaleDisplay__note"
                  >
                     B

                  </h5>
                </div>
              </div>
            </div>

            <div
              class="intervalMap"
            >
              <h4
                class="intervalMap__title"
              >
                Mode interval pattern
              </h4>
              <div
                class="intervalMap__intervals"
              >


                <h5
                  class="intervalMap__interval"
                >
                   tone
                </h5>

                <span
                  class="intervalMap__separator"
                >
                   -
                </span>


                <h5
                  class="intervalMap__interval"
                >
                   tone
                </h5>

                <span
                  class="intervalMap__separator"
                >
                   -
                </span>


                <h5
                  class="intervalMap__interval"
                >
                   semitone
                </h5>

                <span
                  class="intervalMap__separator"
                >
                   -
                </span>


                <h5
                  class="intervalMap__interval"
                >
                   tone
                </h5>

                <span
                  class="intervalMap__separator"
                >
                   -
                </span>


                <h5
                  class="intervalMap__interval"
                >
                   tone
                </h5>

                <span
                  class="intervalMap__separator"
                >
                   -
                </span>


                <h5
                  class="intervalMap__interval"
                >
                   tone
                </h5>

                <span
                  class="intervalMap__separator"
                >
                   -
                </span>


                <h5
                  class="intervalMap__interval"
                >
                   semitone
                </h5>


              </div>
            </div>
            <div
              class="chordMap"
            >
              <h4
                class="chordMap__title"
              >
                Chords (triads)
              </h4>
              <div
                class="chordMap__chords"
              >


                <h5
                  class="chordMap__chord"
                >
                   Cmaj
                </h5>

                <span
                  class="chordMap__separator"
                >
                   -
                </span>


                <h5
                  class="chordMap__chord"
                >
                   Dmin
                </h5>

                <span
                  class="chordMap__separator"
                >
                   -
                </span>


                <h5
                  class="chordMap__chord"
                >
                   Emin
                </h5>

                <span
                  class="chordMap__separator"
                >
                   -
                </span>


                <h5
                  class="chordMap__chord"
                >
                   Fmaj
                </h5>

                <span
                  class="chordMap__separator"
                >
                   -
                </span>


                <h5
                  class="chordMap__chord"
                >
                   Gmaj
                </h5>

                <span
                  class="chordMap__separator"
                >
                   -
                </span>


                <h5
                  class="chordMap__chord"
                >
                   Amin
                </h5>

                <span
                  class="chordMap__separator"
                >
                   -
                </span>


                <h5
                  class="chordMap__chord"
                >
                   Bdim
                </h5>


              </div>
            </div>
          </div>
        </app-fretonator>


        <app-video-loader>
          <div
            class="videoLoader"
          >
            <div
              class="videoLoader__meta"
            >
              <h3
                class="videoLoader__title"
              >
                Try a jam track
              </h3>
              <p
                class="videoLoader__metaDetail"
              >
                 C Ionian (Major) Jam Track by Tom Strahle
              </p>
            </div>
            <div
              class="videoLoader__video"
            >


              <app-video-browser
                class="videoLoader__content"
              >
                <iframe
                  allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
                  allowfullscreen=""
                  frameborder="0"
                  height="100%"
                  src="https://www.youtube.com/embed/oMZz2FeV8q4"
                  title="C Ionian (Major) Jam Track"
                  width="100%"
                />
              </app-video-browser>
            </div>
          </div>
        </app-video-loader>
      </div>
    </main>
  </app-home-index>
</app-home-index-spec>
`;
